File: MasterDirectory/MasterDirectory.Web/Modules/Operaciones/CatalogosOperaciones/CatalogosOperacionesForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.Operaciones.Forms;

[FormScript("Operaciones.CatalogosOperaciones")]
[BasedOnRow(typeof(CatalogosOperacionesRow), CheckNames = true)]
public class CatalogosOperacionesForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}

File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Operaciones/CatalogosOperacionesForm.ts
import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosOperacionesForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosOperacionesForm extends PrefixedContext {
    static formKey = 'Operaciones.CatalogosOperaciones';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosOperacionesForm.init)  {
            CatalogosOperacionesForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosOperacionesForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}